### PR TITLE
DHFPROD-8530: Table view modal for group nodes

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description
RTL tests added. e2e will be added as part of DHFPROD-8556, on top of @csilva-ml 's group nodes related test cases.
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests  

- ##### Reviewer:

- [x] Reviewed Tests
- [x] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/explore/graph-vis-explore/graph-vis-explore.tsx [hunk 1]
```diff
@@ -532,24 +525,30 @@ const GraphVisExplore: React.FC<Props> = (props) => {
...
+          let expandId = payloadData["nodeInfo"] ? payloadData.nodeInfo["expandId"] : clickedNode["expandId"];
+          expandedNodeInfo[expandId] = {
+            nodes: response.data.nodes,
+            edges: response.data.edges,
+            removedNode: payload.data["parentIRI"]
+          };
+          setExpandedNodeData(expandedNodeInfo);
+          network.body.data.nodes.remove(payload.data["parentIRI"]);
+          await updateGroupAndLeafNodesDataset(response.data.nodes);
+          await updateGraphDataWith(response.data.nodes, response.data.edges);
```
Comment (yubiselmarklogic) [inline marklogic-data-hub-central/ui/src/components/explore/graph-vis-explore/graph-vis-explore.tsx:550]: If the that calls don't have dependency in the result the performance can be improved doing something like this ```await Promise.all([updateGroupAndLeafNodesDataset(response.data.nodes),  updateGraphDataWith(response.data.nodes, response.data.edges)])```

Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/explore/graph-vis-explore/graph-vis-explore.tsx:550]: That code is related to leaf nodes (and unrelated to the feature in this PR). 
However, thats a good point, but it seems the second await statement uses the group nodes set during the first one, so thats something we should definitely revisit once the leaf nodes related backend issue is resolved to validate the behavior properly in UI.

**Comment by briantang** [comment]
@xnikhil08 Let's update the 5.7 docs with this new feature: https://wiki.marklogic.com/display/ENGINEERING/Data+Hub+5.7.0

**Review by briantang — DISMISSED** [review]
Functionality looks good and tests passing.

**Review by yubiselmarklogic — APPROVED** [review]

File: marklogic-data-hub-central/ui/src/components/explore/table-view-group-nodes/table-view-group-nodes.tsx [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+import {HCButton} from "@components/common";
+import ResultsTabularView from "@components/results-tabular-view/results-tabular-view";
+import {searchResultsQuery} from "../../../api/queries";
+import {SearchContext} from "../../../util/search-context";
+import {getTableProperties} from "../../../util/data-conversion";
+import {UserContext} from "../../../util/user-context";
+import {expandThresholdExceededWarning, graphViewConfig} from "../../../config/explore.config";
+import {pagePropertiesType} from "types/query-types";
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/explore/table-view-group-nodes/table-view-group-nodes.tsx:15]: Done.

**Review by aniamy — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Pull request merged**